### PR TITLE
Fix #709: Fix issues found by coverity scan

### LINKED ISSUE #709: Fix issues found by coverity scan
Fix or invalidate issues found by coverity scan:

350693 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/attribute/OperationNoteFieldAttribute.java | OperationNoteFieldAttribute.OperationNoteFieldAttribute
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
350690 | Dereference null return value | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java | TppEngineClient.httpStatusException
350689 | Missing authorization check | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Medium impact security | /powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java | AuthMethodController.buildAuthorizationResponse
350685 | Unsafe deserialization | High | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | High impact security | /powerauth-webflow-client/src/main/java/io/getlime/security/powerauth/app/webflow/demo/controller/HomeController.java | HomeController.getApplicationContext
350680 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-data-adapter-model/src/main/java/io/getlime/security/powerauth/lib/dataadapter/model/entity/attribute/AmountAttribute.java | AmountAttribute.AmountAttribute
350675 | Dereference null return value | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-webflow-resources/src/main/java/io/getlime/security/powerauth/lib/webflow/resource/controller/UserProfileController.java | UserProfileController.userInfo
350671 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/attribute/OperationKeyValueFieldAttribute.java | OperationKeyValueFieldAttribute.OperationKeyValueFieldAttribute
350669 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-data-adapter-model/src/main/java/io/getlime/security/powerauth/lib/dataadapter/model/entity/attribute/HeadingAttribute.java | HeadingAttribute.HeadingAttribute
350667 | Dereference null return value | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/OperationPersistenceService.java | OperationPersistenceService.getPendingOperations
350664 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-data-adapter-model/src/main/java/io/getlime/security/powerauth/lib/dataadapter/model/entity/attribute/KeyValueAttribute.java | KeyValueAttribute.KeyValueAttribute
350663 | Dereference null return value | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-webflow-resources/src/main/java/io/getlime/security/powerauth/lib/webflow/resource/controller/UserProfileController.java | UserProfileController.me
350660 | Unsafe dependence on database key | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | API usage errors | /powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/repository/model/entity/OperationAfsActionEntity.java | OperationAfsActionEntity.hashCode
350659 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-data-adapter-model/src/main/java/io/getlime/security/powerauth/lib/dataadapter/model/entity/attribute/NoteAttribute.java | NoteAttribute.NoteAttribute
350652 | Trust boundary violation | Low | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Low impact security | /powerauth-webflow-authentication-login-sca/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/loginsca/controller/LoginScaController.java | LoginScaController.updateUsernameInHttpSession
350643 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/attribute/OperationHeadingFieldAttribute.java | OperationHeadingFieldAttribute.OperationHeadingFieldAttribute
350635 | Unsafe dependence on database key | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | API usage errors | /powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/repository/model/entity/OperationAfsActionEntity.java | OperationAfsActionEntity.equals
350625 | Explicit null dereferenced | Medium | New | 11/12/19 | Unassigned | Unclassified | Unspecified | Undecided | powerauth-webflow | Null pointer dereferences | /powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/attribute/OperationAmountFieldAttribute.java | OperationAmountFieldAttribute.OperationAmountFieldAttribute

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-webflow-authentication-login-sca/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/loginsca/controller/LoginScaController.java [hunk 1]
```diff
@@ -234,6 +234,10 @@ protected AuthMethod getAuthMethodName() {
      */
     private void updateUsernameInHttpSession(String username) {
         synchronized (httpSession.getServletContext()) {
+            if (username == null || !username.matches("^[0-9a-zA-Z_]+$")) {
```
Comment (petrdvorak) [inline powerauth-webflow-authentication-login-sca/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/loginsca/controller/LoginScaController.java]: Maybe username should also be able to contain `@`, `.`, `-`, `/`, `\` or any character that can be typed, to facilitate usernames represented via e-mail, "domain username" (such as `wultra:users/roman`), etc.

**Comment by romanstrobl** [comment]
Validation was requested by Coverity because it considers usernames sensitive. We can also skip the validational and mark the issue as intentional.

**Comment by petrdvorak** [comment]
@romanstrobl Validation is fine, but we need to relax it a bit. For example, to have a max. length, only allow printable characters, etc. The regexp you use does not allow such a simple username as an e-mail address (some banks actually use this format)...

**Comment by romanstrobl** [comment]
Am I missing something? `^[a-zA-Z0-9_\-@\./\\:;<>!#\$%&'"\*+=\?\^`\(\)\{\}\[\]|~]{4,32}$`

**Comment by petrdvorak** [comment]
This looks good, I would extend the length to 256 max...

**Comment by romanstrobl** [comment]
I updated the validation regexp as requested.

**Pull request merged**